Tools/Scripts/compare-static-analysis-results

```diff
@@ -332,8 +341,35 @@ def main():
         if type_total:
             print(f'Total {type}: {len(type_total)}')
 
-    if args.report_urls:
-        upload_results(args, results_for_upload)
+    return results_for_upload
+
+
+def generate_results_only(args):
+    result_paths = []
+    print(f'Generating new results index with results from {args.build_output}...')
+
+    with open(os.path.abspath(f"{args.build_output}/unexpected_results.json"), "r") as f:
+        results_data = json.load(f)
+    with open(f'{args.new_dir}/issues_per_file.json') as f:
+        issues_per_file = json.load(f)
+
+    for project, checkers in results_data.get('failures').items():
+        for checker, files in checkers.items():
+            for file in files:
+                result_paths += [i for i in issues_per_file[checker][file].values()]
+
+    create_filtered_results_dir(args, project, result_paths, category='StaticAnalyzerRegressions')
+
+
+def main():
+    args = parser()
+
+    if not args.generate_results:
```

Comment:
It's confusing that this variable is called "generate results". We're always generating results.
Maybe generate_filtered_results?